packages/xpetra/src/MultiVector/Xpetra_MultiVector_def.hpp

```diff
@@ -89,6 +89,26 @@ Xpetra_randomize()
     }
 }
 
+template<class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
+void
+MultiVector<Scalar, LocalOrdinal, GlobalOrdinal, Node>::
+Xpetra_randomize(const Scalar& minVal, const Scalar& maxVal)
+{
+    typedef Teuchos::ScalarTraits<Scalar> SCT;
+
+    const size_t numVectors = getNumVectors();
+    for(size_t i = 0; i < numVectors; i++)
+    {
+        Teuchos::ArrayRCP<Scalar> datai = getDataNonConst(i);
+
+        const size_t myLength = getLocalLength();
+        for(size_t j = 0; j < myLength; j++)
+        {
+          datai[ j ] = 0.5*(maxVal-minVal)*SCT::random()+0.5*(maxVal+minVal);
```

Comment:
Is this really what you want?  See https://github.com/trilinos/Trilinos/blob/7527310aa8d57d922fca98bb3c0ea42a1ecf53bb/packages/ml/src/Utils/ml_utils.c#L1825.